message/data.go

```diff
@@ -1,76 +1,187 @@
+// message defines the received Json messages and their nested fields
 package message
 
+import (
+	"encoding/json"
+)
+
```

Comment:
Also put the same comment about concatenation here ? 
(Yeah, I'm really scared this information will get lost to next semester team otherwise)
// All concatenation for hashes mentionned in this document are based on the following concept
["item1", "item2"]